### ISSUE TITLE
Use class instead of className for unidentified DOM elements (better support for styling web components)

### ISSUE THREAD

**Issue by seanaguinaga** [stitchesjs/stitches#654]
I'd like stitches to use class instead of className when components are not recognized or are web-components.

Please watch video for demo of issue/request (2 mins long)

[Video](https://sean-hey.tinytake.com/tt/NTU1Nzg3Ml8xNzMxNjQxOQ)

https://github.com/styled-components/styled-components/blob/main/CHANGELOG.md

Styled Components should release this in v6 and there might be more information about this styling issue in their repo.

**Comment by peduarte** [comment]
Hi Sean, thanks for the video. That was super helpful 🙏 

We'd need to look into this. Also, we'd need to experiment with different approaches and ensure it doesn't affect performance and/or bundle size too much - if we decide to go ahead.

How do you see this working? 
- You explicitly tell Stitches that it's a web component?
- Happens seamlessly, the library "knows" its a web component and uses `class` instead?

Related: https://reactjs.org/docs/dom-elements.html#classname

**Comment by seanaguinaga** [comment]
- **Happens seamlessly, the library "knows" its a web component and uses class instead?**

I would have it check this list:

[domElements](https://github.com/styled-components/styled-components/blob/main/packages/styled-components/src/utils/domElements.ts)

and switch to class if not found.

Which is an approach others use: (I asked via twitter)

https://twitter.com/seanaguinaga/status/1409614205389459469?s=20

**Comment by jonathantneal** [comment]
We talked about making the `className` property configurable, and that might be my recommendation here.

As far as the implementation expectations — those I’m concerned with.

The cited [list of elements](https://github.com/styled-components/styled-components/blob/main/packages/styled-components/src/utils/domElements.ts) is used to ["handle custom elements which React doesn't properly alias"](https://github.com/styled-components/styled-components/blob/8165cbe994f6f749236244f6f7017c2f0b9afcfe/packages/styled-components/src/models/StyledComponent.ts#L157). Why does React not properly alias custom elements? That seems like a problem. If it’s a problem, might this break once React fixes it? Or is this a (_probably unhelpful, but_) deliberate decision? And are we sure React has problems with _just_ custom elements?

And are all property aliases ignored, or is this limited to `class` / `className`? For what it’s worth, React already does not properly alias a long list of properties (like the ARIA reflection _properties_ (**not attributes**)).

That list also reflects ["all valid HTML5 elements"](https://github.com/styled-components/styled-components/blob/30dab74acedfd26d227eebccdcd18c92a1b3bd9b/packages/styled-components/src/constructors/test/styled.test.ts#L6), but which release of HTML5 is this? I do see some HTML elements in there, but also some SVG elements, and no MathML elements. I see no HTML `template` element or HTML `slot` element in there, either. Did they happen to forget `template`? I want to doubt this. Instead, does React use a similar _hand-wavy_ list internally that’s been copied by SC?

If we were to detect [custom elements](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements) and swap property names accordingly, why not check for a hyphen dash (`-`) instead of a long, ever-changing list of elements? A dash is required for custom elements. In the video, the dash is properly used.

In summary, the request by @seanaguinaga seems super fair to consider 👍, but the implementation details leave me with a lot of concerns.

**Comment by seanaguinaga** [comment]
Configurability seems like a great solution to prevent any unintended bugs.

Do you see that being on a per 

```
styled("ion-button" , { webComponent: true});
```

kind of a thing?

or in the stitches config?

**Comment by seanaguinaga** [comment]
Another way would be to use the most comprehensive/up to date list of HTML5 Elements

**Comment by peduarte** [comment]
Thanks Sean. Will have a think. We're currently working on the TypeScript layer so will likely wait for it to finish before looking into this.

**Comment by seanaguinaga** [comment]
@peduarte did the TypeScript layer end up getting finished?
